runtime/moonbase/src/migrations.rs

```diff
@@ -0,0 +1,75 @@
+//TODO license
+
+//! State migrations for the Moonbase runtime
+
+use frame_support::{traits::{Get, OnRuntimeUpgrade, ReservableCurrency}, weights::Weight};
+use crate::{Balance, Runtime, AuthorMapping};
+
+/// The type responsible for migrating the moonbase runtime from ersion to version. Each migration
+/// is gated by runtime spec version, so only the correct migrations should be supplied. If you
+/// need to migrate storage, add the migration logic here.
+pub struct MoonbaseMigrationStation;
+
+impl OnRuntimeUpgrade for MoonbaseMigrationStation {
+	fn on_runtime_upgrade() -> Weight {
+
+		log::info!("🗺️ Running MoonbaseMigrationStation");
+
+        //TODO I'd like to detect the correct version that we ar emigrating _from_ here. This code
+        // seems to always give me the same version 43 from _this_ runtime though. Shawn's comments
+        // sound like that is expected.
+		// let previous_version: u32 = frame_system::LastRuntimeUpgrade::<Runtime>::get().map(|rv| rv.spec_version.into()).unwrap_or(0);
+        let previous_version = 40;
+
+		log::info!("🗺️ Previous version: {:?}", previous_version);
+		log::info!("🗺️ Upgrade version:  {:?}", crate::VERSION.spec_version);
+
+		match previous_version {
+			v if v < 40 => panic!("Migrating from runtimes below spec_version 40 is not supported!"),
+			v if v < 43 => migrate_author_mapping_storage_deposit_to_proper_units(),
+			_ => panic!("Cannot migrate from future runtime versions"),
+
+		}
+	}
+}
+
+fn migrate_author_mapping_storage_deposit_to_proper_units() -> Weight {
```

Comment:
Would it make sense to have some very explicit versioning in this fn name? I'm envisioning a future where we have many migration paths and our migration code may come with a very large maintenance cost. Something like `migrate_author_mapping_pre_X_to_X()`, etc.

Reply:
Nevermind, I hadn't caught up on the conversation before writing this; I think I had the wrong mindset.